File: DRCHEF.py
'''

* Author : Ayushman Chahar #
* About  : II Year, IT Undergrad #
* Insti  : VIT, Vellore #

'''

import os
import sys
import bisect
from io import BytesIO, IOBase

BUFSIZE = 8192


class FastIO(IOBase):
    newlines = 0

    def __init__(self, file):
        self._fd = file.fileno()
        self.buffer = BytesIO()
        self.writable = "x" in file.mode or "r" not in file.mode
        self.write = self.buffer.write if self.writable else None

    def read(self):
        while True:
            b = os.read(self._fd, max(os.fstat(self._fd).st_size, BUFSIZE))
            if not b:
                break
            ptr = self.buffer.tell()
            self.buffer.seek(0, 2), self.buffer.write(b), self.buffer.seek(ptr)
        self.newlines = 0
        return self.buffer.read()

    def readline(self):
        while self.newlines == 0:
            b = os.read(self._fd, max(os.fstat(self._fd).st_size, BUFSIZE))
            self.newlines = b.count(b"\n") + (not b)
            ptr = self.buffer.tell()
            self.buffer.seek(0, 2), self.buffer.write(b), self.buffer.seek(ptr)
        self.newlines -= 1
        return self.buffer.readline()

    def flush(self):
        if self.writable:
            os.write(self._fd, self.buffer.getvalue())
            self.buffer.truncate(0), self.buffer.seek(0)


class IOWrapper(IOBase):
    def __init__(self, file):
        self.buffer = FastIO(file)
        self.flush = self.buffer.flush
        self.writable = self.buffer.writable
        self.write = lambda s: self.buffer.write(s.encode("ascii"))
        self.read = lambda: self.buffer.read().decode("ascii")
        self.readline = lambda: self.buffer.readline().decode("ascii")


sys.stdin, sys.stdout = IOWrapper(sys.stdin), IOWrapper(sys.stdout)
input = lambda: sys.stdin.readline().rstrip("\r\n")


def main():
    for _ in range(int(input())):
        n, k = map(int, input().split())
        arr = [int(i) for i in input().split()]
        if (k >= max(arr)):
            print(n)
            continue
        arr.sort()
        maxarr, minarr = arr[-1], arr[0]
        if (k <= minarr):
            days = 0
            meds = k
            temparr = arr
            start = 0
            while (True):
                if (temparr[-1] == 0):
                    break
                if meds >= arr[start]:
                    meds = arr[start]
                    arr[start] = 0
                    start += 1
                    meds *= 2
                    days += 1
                else:
                    days += 1
                    meds *= 2
            print(days)
            continue
        days = bisect.bisect_left(arr, k)
        td1, td2 = 0, 1
        tempk = k
        while(1):
            if tempk >= arr[days]:
                td1 += 1
                break
            else:
                td1 += 1
                tempk *= 2
        tempk = arr[days - 1] * 2
        while(1):
            if tempk >= arr[days]:
                td2 += 1
                break
            else:
                td2 += 1
                tempk *= 2
        if(td2 <= td1):
            days -= 1
        # if(arr[days - 1] * 2 >= arr[days]) and (k != arr[days]):
        #     days -= 1
        # if(arr[days] % arr[days - 1] == 0) and (arr[days] // arr[days - 1] > 2) and (arr[days] != 1 and arr[days - 1] != 1):
        #     days -= 1
        left_pop = arr[days:]
        start = 0
        meds = k
        while (True):
            if (left_pop[-1] == 0):
                break
            if meds >= left_pop[start]:
                meds = left_pop[start]
                left_pop[start] = 0
                start += 1
                meds *= 2
                days += 1
            else:
                days += 1
                meds *= 2
        print(days)

if __name__ == "__main__":
    main()

File: DRCHEF.cpp
/*

* Author : Ayushman Chahar #
* About  : II Year, IT Undergrad #
* Insti  : VIT, Vellore #

*/

//Directive, Optimisation Flags, Targets
#pragma GCC optimize("Ofast")  
#pragma GCC target("avx,avx2,fma") 
#pragma GCC optimization ("unroll-loops")

//Header Files
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <sstream>
#include <queue>
#include <deque>
#include <bitset>
#include <iterator>
#include <list>
#include <stack>
#include <map>
#include <set>
#include <functional>
#include <numeric>
#include <utility>
#include <limits>
#include <time.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
//#include <boost/multiprecision/cpp_int.hpp>

//using boost :: multiprecision :: cpp_int;
using namespace std;
using std :: string;

//Pre-Processors and TypeDef constants
#define SCD(t) scanf("%d",&t)
#define SCLD(t) scanf("%ld",&t)
#define SCLLD(t) scanf("%lld",&t)
#define SCC(t) scanf("%c",&t)
#define SCS(t) scanf("%s",t)
#define SCF(t) scanf("%f",&t)
#define SCLF(t) scanf("%lf",&t)
#define MEM(a, b) memset(a, (b), sizeof(a))
#define FOR(i, j, k, in) for (int i=j ; i<k ; i+=in)
#define RFOR(i, j, k, in) for (int i=j ; i>=k ; i-=in)
#define REP(i, j) FOR(i, 0, j, 1)
#define RREP(i, j) RFOR(i, j, 0, 1)
#define all(cont) cont.begin(), cont.end()
#define rall(cont) cont.end(), cont.begin()
#define FOREACH(it, l) for (auto it = l.begin(); it != l.end(); it++)
#define IN(A, B, C) assert( B <= A && A <= C)
#define MP make_pair
#define PB push_back
#define INF (int)1e9
#define EPS 1e-9
#define PI 3.1415926535897932384626433832795
#define MOD 1000000007
#define read(type) readInt<type>()
#define McQueen ios_base::sync_with_stdio(false);cin.tie(nullptr);cout.tie(nullptr)
const double pi = acos(-1.0);
typedef long long int lli;
typedef unsigned long long int ulli;
typedef long int li;
typedef unsigned long int uli;


//Basic min and max
template<typename T, typename U> inline void amin(T &x, U y) { if(y < x) x = y; }
template<typename T, typename U> inline void amax(T &x, U y) { if(x < y) x = y; }

/**********************************************/

// Fast I/O Methods
template <typename T> inline void write(T x)
{
    int i = 20;
    char buf[21];
    // buf[10] = 0;
    buf[20] = '\n';
    do
    {
        buf[--i] = x % 10 + '0';
        x /= 10;
    }while(x);
    do
    {
        putchar(buf[i]);
    } while (buf[i++] != '\n');
}

template <typename T> inline T readInt()
{
    T n = 0,s = 1;
    char p = getchar();
    if(p == '-')
        s =- 1;
    while((p < '0' | p > '9') && p != EOF && p != '-')
        p = getchar();
    if(p == '-')
        s =- 1,p = getchar();
    while(p >= '0' && p <= '9') {
        n = (n << 3) + (n << 1) + (p - '0');
        p = getchar();
    }

    return n * s;
}

/************************************/

// Let's Debug!
#define DEBUG

#ifdef DEBUG

    #define debug(args...)     (Debugger()) , args

    class Debugger
    {
        public:
        Debugger(const std::string& _separator = " - ") :
        first(true), separator(_separator){}

        template<typename ObjectType> Debugger& operator , (const ObjectType& v)
        {
            if(!first)
                std:cerr << separator;
            std::cerr << v;
            first = false;
            return *this;
        }
        ~Debugger() {  std:cerr << endl;}

        private:
        bool first;
        std::string separator;
    };

#else
    #define debug(args...)                  // Just strip off all debug tokens
#endif

/**************************************/

//Initialize main()
int main() 
{
    //McQueen;
    int t;
    cin >> t;
    while(t--) {
    	int n, k;
    	cin >> n >> k;
    	vector<int> arr(n);
    	for(int& i: arr) cin >> i;
    	if(k >= *max_element(arr.begin(), arr.end())) {
    		cout << n << "\n";
    		continue;
    	}
    	sort(arr.begin(), arr.end());
    	if(k <= arr[0]) {
    		int days = 0, start = 0;
    		lli meds = k;
    		while(1) {
    			if(arr[n - 1] == 0) break;
    			if(meds >= arr[start]) {
    				meds = arr[start];
    				arr[start] = 0;
    				start++;
    				days++;
    				meds *= 2;
    			} else {
    				days++;
    				meds *= 2;
    			}
    		}
    		cout << days << "\n";
    		continue;
    	}
    	auto it = lower_bound(arr.begin(), arr.end(), k);
    	int days = it - arr.begin();
    	int td1 = 0, td2 = 1;
    	lli tempk = k;
    	while(1) {
    		if(tempk >= arr[days]) {
    			td1++;
    			break;
    		} else {
    			td1++;
    			tempk *= 2;
    		}
    	}
    	tempk = arr[days - 1] * 2;
    	while(1) {
    		if(tempk >= arr[days]) {
    			td2++;
    			break;
    		} else {
    			td2++;
    			tempk *= 2;
    		}
    	}
    	if(td2 <= td1) days--;
    	vector<int> left_pop(arr.begin() + days, arr.end());
    	int start = 0,size = left_pop.size();
    	lli meds = k;
    	while(1) {
    		if(left_pop[size - 1] == 0) break;
    		if(meds >= left_pop[start]) {
    			meds = left_pop[start];
    			left_pop[start] = 0;
    			start++;
    			days++;
    			meds *= 2; 
    		} else {
    			days++;
    			meds *= 2;
    		}
    	}
    	cout << days << "\n";
    }
    return 0;
}

